beetsplug/edit.py

```diff
@@ -324,3 +333,55 @@ def save_changes(self, objs):
             if ob._dirty:
                 self._log.debug('saving changes to {}', ob)
                 ob.try_sync(ui.should_write(), ui.should_move())
+
+    # Methods for interactive importer execution.
+
+    def before_choose_candidate_listener(self, session, task):
+        """Append an "Edit" choice and an "edit Candidates" choice (if
+        there are candidates) to the interactive importer prompt.
+        """
+        choices = [PromptChoice('d', 'eDit', self.importer_edit)]
+        if task.candidates:
```

Comment:
This one actually revealed a problem if the user uses `Enter search` or `enter Id` to find a candidate not found during the initial lookup: as things stand, I believe this listener has no way of retrieving that candidate (as the new `candidate` it is handled *within* `TerminalImportSession.choose_match()`, and only saved when exiting the prompt by applying). I have added a note to the docs, but I'm wondering if something else should be done about it.